src/v/container/fragmented_vector.h

```diff
@@ -52,21 +52,24 @@ class fragmented_vector {
 
     // calculate the maximum number of elements per fragment while
     // keeping the element count a power of two
-    static constexpr size_t calc_elems_per_frag(size_t esize) {
-        size_t max = fragment_size_bytes / esize;
+    static constexpr size_t calc_elems_per_frag() {
+        size_t max = fragment_size_bytes / sizeof(T);
         if constexpr (is_chunked_vector) {
-            max = max_allocation_size / esize;
+            max = max_allocation_size / sizeof(T);
         }
-        assert(max > 0);
         return std::bit_floor(max);
     }
 
-    static constexpr size_t elems_per_frag = calc_elems_per_frag(sizeof(T));
-
-    static_assert(
-      (elems_per_frag & (elems_per_frag - 1)) == 0,
-      "element count per fragment must be a power of 2");
-    static_assert(elems_per_frag >= 1);
+    /**
+     * The maximum number of bytes per fragment as specified in
+     * as part of the type. Note that for most types, the true
+     * number of bytes in a full fragment may as low as half
+     * of this amount (+1) since the number of elements is restricted
+     * to a power of two.
```

Comment:
is it still retstricted to power of two after this commit?

Reply:
Yes.